platform/linux-generic/odp_stash.c

```diff
@@ -369,61 +492,59 @@ static inline int32_t stash_put(odp_stash_t st, const void *obj, int32_t num)
 	obj_size = stash->obj_size;
 
 	if (obj_size == sizeof(uint64_t)) {
-		ring_u64_t *ring_u64 = &stash->ring_u64.hdr;
+		if (batch)
+			return stash->ring_fn.u64.enq_batch(stash, (uint64_t *)(uintptr_t)obj, num);
 
-		ring_u64_enq_multi(ring_u64, stash->ring_mask,
-				   (uint64_t *)(uintptr_t)obj, num);
-		return num;
+		return stash->ring_fn.u64.enq_multi(stash, (uint64_t *)(uintptr_t)obj, num);
 	}
 
 	if (obj_size == sizeof(uint32_t)) {
-		ring_u32_t *ring_u32 = &stash->ring_u32.hdr;
+		if (batch)
+			return stash->ring_fn.u32.enq_batch(stash, (uint32_t *)(uintptr_t)obj, num);
 
-		ring_u32_enq_multi(ring_u32, stash->ring_mask,
-				   (uint32_t *)(uintptr_t)obj, num);
-		return num;
+		return stash->ring_fn.u32.enq_multi(stash, (uint32_t *)(uintptr_t)obj, num);
 	}
 
 	if (obj_size == sizeof(uint16_t)) {
 		const uint16_t *u16_ptr = obj;
-		ring_u32_t *ring_u32 = &stash->ring_u32.hdr;
 		uint32_t u32[num];
 
 		for (i = 0; i < num; i++)
 			u32[i] = u16_ptr[i];
 
-		ring_u32_enq_multi(ring_u32, stash->ring_mask, u32, num);
-		return num;
+		if (batch)
+			return stash->ring_fn.u32.enq_batch(stash, u32, num);
+
+		return stash->ring_fn.u32.enq_multi(stash, u32, num);
 	}
 
 	if (obj_size == sizeof(uint8_t)) {
 		const uint8_t *u8_ptr = obj;
-		ring_u32_t *ring_u32 = &stash->ring_u32.hdr;
 		uint32_t u32[num];
 
 		for (i = 0; i < num; i++)
 			u32[i] = u8_ptr[i];
 
-		ring_u32_enq_multi(ring_u32, stash->ring_mask, u32, num);
-		return num;
+		if (batch)
+			return stash->ring_fn.u32.enq_batch(stash, u32, num);
+
+		return stash->ring_fn.u32.enq_multi(stash, u32, num);
 	}
 
 	return -1;
 }
 
 int32_t odp_stash_put(odp_stash_t st, const void *obj, int32_t num)
 {
-	return stash_put(st, obj, num);
+	return stash_put(st, obj, num, 0);
```

Comment:
The batch identifier is `odp_bool_t` so could use `true`/`false`.

Reply:
Fixed on V2.